jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/WebSocketUpgradeHandler.java

```diff
@@ -116,4 +116,13 @@ public boolean handle(Request request, Response response, Callback callback) thr
             return true;
         return super.handle(request, response, callback);
     }
+
+    @Override
+    public InvocationType getInvocationType()
+    {
+        // Must be BLOCKING because EndPoint.upgrade() ends up invoking
+        // application code in the WebSocket "connect" event handler,
+        // e.g. a method annotated with @OnWebSocketConnect, that may block.
+        return InvocationType.NON_BLOCKING;
```

Comment:
comment doesn't match the return code?

Reply:
Can we execute the application code or the upgrade?
Otherwise if we ever use websocket, then we can never be non-blocking

Reply:
Doh!

Reply:
We do not want to execute the WebSocket listener (or annotated) methods, otherwise we're going to sink the performance.

I have captured the non-blocking problem in #9552 -- I don't think there is a way around with the current APIs.
If we mark `WebSocketUpgradeHandler` as `NON_BLOCKING`, then we block the selector thread.
If we impose that WebSocket listener (or annotated) methods _must_ be non-blocking, then we either force application to be careful and _not_ use the blocking APIs that we offer, and/or we force applications to always use `suspend()/resume()` -- hence #9552, where we only offer non-blocking APIs, an explicit demand API, we impose non-blocking behavior and if the application code blocks on other APIs, then we block the selector thread, but at least it's clear and documented.

Note that right now, for a setup with just a `ContextHandler`, we block the selector thread if the application `Handler` (child of `ContextHandler`) blocks on other APIs.
At least we would have a consistent behavior.

Reply:
Then we need to at least make this configurable, so an application that uses websocket can promise not to block and thus configure this handler to be non-blocking.  Otherwise it is a bit sad that we are saying we can NEVER have a completely asynchronous server+application that uses websocket!

Reply:
`WebSocketUpgradeHandler` inherits `setDynamic(boolean)` which is typically used to mark the `Handler` as `BLOCKING`.

The name `setDynamic()` does not really hint at the `InvocationType`, but would do the job.
Perhaps rename to `setBlocking()` (as we don't really want to expose `EITHER`)?